archive_communication/api-requirements.md

```diff
@@ -0,0 +1,324 @@
+# API Requirements
+
+This document outlines some general requirements for APIs that would be necessary
+for communication between BCDC, data archives, and other partners for one- or
+two-way communication. These requirements don't mandate a specific implementation, 
+but do make some assumptions around higher level design:
+
+## Assumptions
+- This document and the examples make a few assumptions about the APIs:
+    - APIs communicate via JSON over HTTP
+    - APIs roughly follow REST principles
+
+## General Requirements
+
+<table class="tg">
+<thead>
+  <tr>
+    <th class="tg-g7sd">#</th>
+    <th class="tg-g7sd">Requirement</th>
+    <th class="tg-g7sd">Description</th>
+    <th class="tg-g7sd">Priority</th>
+    <th class="tg-g7sd">Notes</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td class="tg-lboi">1</td>
+    <td class="tg-lboi">Individual user authentication</td>
+    <td class="tg-lboi">If the API requires authentication, an individual user can authenticate themselves with an API Key or OAuth access token to interact with the API.</td>
+    <td class="tg-lboi">Should have</td>
+    <td class="tg-lboi">
+        <ul>
+            <li>Preferred OAuth mechanism would be ORCID.</li>
+        </ul>
+    </td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">2</td>
+    <td class="tg-lboi">Service account authentication</td>
+    <td class="tg-lboi">A downstream service can authenticate with an API Key to automate interactions with the API.</td>
+    <td class="tg-lboi">Must have</td>
+    <td class="tg-lboi"></td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">3</td>
+    <td class="tg-lboi">Consistent, persistent identifier for objects</td>
+    <td class="tg-lboi">A user or service should be able to access an object using the same identifying URI each time.</td>
+    <td class="tg-lboi">Must have</td>
+    <td class="tg-lboi"></td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">4</td>
+    <td class="tg-lboi">Object schema documentation</td>
+    <td class="tg-lboi">Objects fetched from API contain links to the schema used to generate/validate objects.</td>
+    <td class="tg-lboi">Must have</td>
+    <td class="tg-lboi">
+        <ul>
+            <li>JSONSchema preferred</li>
+        </ul>
+    </td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">5</td>
+    <td class="tg-lboi">Versioned data schemas</td>
+    <td class="tg-lboi">Schemas should have versions so as the schemas change, we can track which schema was used to generate/validate objects.</td>
+    <td class="tg-lboi">Should have</td>
+    <td class="tg-lboi">
+        <ul>
+            <li>JSONSchema preferred</li>
+        </ul>
+    </td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">6</td>
+    <td class="tg-lboi">Related objects are easily findable from a given object</td>
```

Comment:
@pbishwakarma you mean the response for each object should contain information about parents and siblings? E.g. relationships between datasets and collections?